### PR TITLE
UNR-2507 - LoadObjects when bNetLoadOnClient is false and object is deleted.

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
If an actor had `bNetLoadOnClient=false` and all references to the actor were deleted, it is not possible to use FindObject to reload that actor. We have to use LoadObject. I double-checked and we match native behaviour here in regards to instanced data so no concerns there. 

#### Release note
Bugfix: Actors placed in the level with bNetLoadOnClient=false can now be reloaded once deleted.

#### Tests
Set up a test gym with actors which had bNetLoadOnClient=true and bNetLoadOnClient=false. Both were reloaded correctly with the fix, instanced data scenarios matched native.

#### Primary reviewers
@m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealMetadata.h [hunk 1]
```diff
@@ -85,7 +85,9 @@ struct UnrealMetadata : Component
 		{
 			Class = FindObject<UClass>(nullptr, *ClassPath, false);
 		}
-		else
+
+		// It's possible we were unable to load a stably named actor as all references have been deleted.
+		if (!Class)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealMetadata.h]: Just so we're on the same page - does that mean that we believe the object is stably named, but unreal doesn't? If Unreal did, we wouldn't be deleting it when it goes out of scope.

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealMetadata.h]: True, we have a `StablyNamedRef` and `Object->IsFullNameStableForNetworking()` is false.

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/UnrealMetadata.h]: Stably named refs are given to these placed actors due to this PR: https://github.com/spatialos/UnrealGDK/pull/721
```
	// We want to have a stably named ref if this is a loaded Actor.
	// We use this to indicate if a new Actor should be created or to link a pre-existing Actor when receiving an AddEntityOp.
	// Previously, IsFullNameStableForNetworking was used but this was only true if bNetLoadOnClient was true.
```
Looks like it's to help with server re-connection.

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Fixed an crash caused by attempting to read schema from an unloaded class.
 - Unresolved object references in replicated arrays of structs should now be properly handled and eventually resolved.
 - Fix tombstone-related assert that could fire and bring down the editor.
+- Actors placed in the level with bNetLoadOnClient=false can now be reloaded once all references are deleted.
```
Comment (m-samiec) [inline CHANGELOG.md]:
```suggestion
- Actors placed in the level with bNetLoadOnClient=false that go out of view will now be reloaded if they come back into view.
```

**Pull request merged**